extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -141,6 +141,8 @@ export class AmpStoryPage extends AMP.BaseElement {
     /** @private {!Array<function()>} */
     this.unlisteners_ = [];
 
+    /** @private @const {!../../../src/service/timer-impl.Timer} */
+    this.timer_ = Services.timerFor(this.win);
```

Comment:
I was also thinking of potentially using a mutation observer instead in case there in slow down with the animations or the CSS is eventually changed. I was unsure of the browser support and the practicality of the observing/disconnecting the observer, but if you think that (or an entirely different approach) would work better then please let me know!

Reply:
A timer is great :)